runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -11949,11 +11949,11 @@ void J9::Z::TreeEvaluator::genWrtbarForArrayCopy(TR::Node *node, TR::Register *s
    }
 
 TR::Register*
-J9::Z::TreeEvaluator::VMinlineCompareAndSwap(TR::Node *node, TR::CodeGenerator *cg, TR::InstOpCode::Mnemonic casOp, bool isObj)
+J9::Z::TreeEvaluator::VMinlineCompareAndSwap(TR::Node *node, TR::CodeGenerator *cg, TR::InstOpCode::Mnemonic casOp, bool isObj, bool isExchange)
    {
    TR::Register *scratchReg = NULL;
    TR::Register *objReg, *oldVReg, *newVReg;
-   TR::Register *resultReg = cg->allocateRegister();
+   TR::Register *resultReg = isExchange ? NULL : cg->allocateRegister();
```

Comment:
Can we add bracket around the conditions?

Reply:
Actually this is good, no change needed

Reply:
I can do that.

Reply:
Or not, since I saw your new comment.

Reply:
yes, not need to make change, This is good.